presto-main/src/main/java/com/facebook/presto/operator/scalar/sql/MapSqlFunctions.java

```diff
@@ -57,6 +57,17 @@ public static String mapTopNKeysComparator()
         return "RETURN IF(n < 0, fail('n must be greater than or equal to 0'), slice(reverse(array_sort(map_keys(input), f)), 1, n))";
     }
 
+    @SqlInvokedScalarFunction(value = "map_top_n_keys_by_value", deterministic = true, calledOnNullInput = false)
```

Comment:
Not sure if this adds any additional use case because users can themselves run map_keys(map_top_n

Reply:
that's right its based on that but currently we have received feedback that people confuse map_top_n_keys as to produce top n kets of map based on its value but that's not correct. The existing implementation of map_top_n_keys is comparing its keys to get top n keys. 

To overcome this issue, we are releasing map_top_n_keys_by_value and possibly we will deprecate map_top_n_keys later on.